wolfcrypt/src/cmac.c

```diff
@@ -230,9 +230,16 @@ int wc_CmacFinal(Cmac* cmac, byte* out, word32* outSz)
     const byte* subKey;
     word32 remainder;
 
-    if (cmac == NULL || out == NULL || outSz == NULL) {
+    if (cmac == NULL)
         return BAD_FUNC_ARG;
+
+    if (out == NULL || outSz == NULL) {
+        if ((out == NULL) ^ (outSz == NULL))
+            return BAD_FUNC_ARG;
+        ret = 0;
+        goto out;
```

Comment:
In my estimate it wasn't worth the time and risk, and general disruption of incumbent code.  The "goto out" code pattern is used all over our codebase and is even MISRA-compliant (a carve-out in the general no-goto rule).

Reply:
Thanks @philljj. Coding standard lists to avoid goto's if possible. Think we have been to lax on enforcing that.

Because it exists already elsewhere in the code is not a good reason to allow more of it.....though adding unnecessary complexity to the point of being unreasonable without the use of goto's is....